src/storage/txn/scheduler.rs

```diff
@@ -882,6 +887,13 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
 
         self.on_acquired_locks_finished(metadata, new_acquired_locks);
 
+        if !known_txn_status.is_empty() {
```

Comment:
The `result` should be checked here as the cache should be updated only when the `async-write` finishes successfully. Unit-test to cover this path is also needed.